File: pe-gui/src/main/java/org/saar/gui/style/property/Length.java
package org.saar.gui.style.property;

public class Length {

    protected int value = 0;

    public void set(int value) {
        this.value = value;
    }

    public void add(int value) {
        set(get() + value);
    }

    public void sub(int value) {
        set(get() - value);
    }

    public void set(Length length) {
        set(length.get());
    }

    public int get() {
        return this.value;
    }
}

File: pe-gui/src/main/kotlin/org/saar/gui/style/length/Length.kt
package org.saar.gui.style.length

abstract class Length(default: LengthValue = LengthValues.fit) : ReadonlyLength {

    var value: LengthValue = default

    fun set(value: LengthValue) {
        this.value = value
    }

    fun set(pixels: Int) {
        this.value = LengthValues.pixels(pixels)
    }
}